realm/gradle/wrapper/gradle-wrapper.properties

```diff
@@ -4,3 +4,4 @@ distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
 distributionUrl=https\://services.gradle.org/distributions/gradle-2.7-bin.zip
+set JAVA_OPTS="-Xmx512m"
```

Comment:
have you tried the following inside `gradle.properties`
```groovy
org.gradle.jvmargs=-Xms256m -Xmx1024m
```
more here https://docs.gradle.org/current/userguide/build_environment.html